packages/ckeditor5-dev-release-tools/lib/tasks/commitandtag.js

```diff
@@ -39,6 +48,40 @@ export default async function commitAndTag( { version, files, cwd = process.cwd(
 		return;
 	}
 
-	await git.commit( `Release: v${ version }.${ skipCi ? ' [skip ci]' : '' }`, filePathsToAdd );
-	await git.addTag( `v${ version }` );
+	if ( dryRun ) {
+		await executePreCommitHook( { filePathsToAdd, cwd } );
+		listrTask.output = `[dry run] Creating git tag v${ tagForVersion }.`;
+	} else {
+		await git.commit( `Release: v${ version }.${ skipCi ? ' [skip ci]' : '' }`, filePathsToAdd );
+		await git.addTag( `v${ version }` );
+	}
+}
+
+/**
+ * Function that imitates running dry run for the `git commit` command.
+ *
+ * @param {object} options
+ * @param {Array.<string>} options.filePathsToAdd File paths to run the dry run validations for.
+ * @param {string} options.cwd Current working directory from which all paths will be resolved.
+ * @returns {Promise<void>}
+ */
+async function executePreCommitHook( { filePathsToAdd, cwd } ) {
+	const normalizedCwd = toUnix( cwd );
+	const git = simpleGit( { baseDir: normalizedCwd } );
+
+	try {
+		const preDryRunCommit = await git.log( [ '-1' ] );
+		await git.commit( '[dry run] Release.', filePathsToAdd );
+
+		try {
+			await git.reset( [ preDryRunCommit.latest.hash ] );
```

Comment:
I decided to throw an error with a message informing about a potential potential problem with continuing the release in case `git reset` fails during a dry run.

I'm not sure if it's needed, as it probably will not ever happen.